{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "509cf1cc-44c3-4d4e-8489-9458cf0f4c44",
   "metadata": {},
   "source": [
    "# Blast taurine transporter against genomes to check if present"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f35703c3-7813-4851-b321-605167f17b1b",
   "metadata": {},
   "source": [
    "### Make a genome database for each symbiont group and blast tauABC genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ae7b104b-9d4c-424b-9699-d2b0f18b5c47",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash -l\n",
    "#SBATCH --nodes=1\n",
    "#SBATCH --ntasks-per-node=1\n",
    "#SBATCH --cpus-per-task=20\n",
    "#SBATCH --mem=50G\n",
    "#SBATCH --job-name=blast\n",
    "#SBATCH --time=6:00:00\n",
    "#SBATCH --partition=general\n",
    "#SBATCH --account=a_ace\n",
    "#SBATCH -o /home/uqpobri2/scripts/out/blast.output\n",
    "#SBATCH -e /home/uqpobri2/scripts/error/blast.error\n",
    "\n",
    "##########################\n",
    "\n",
    "# use blast to search taurine transporter against symbiont genomes (reviewer comment)\n",
    "\n",
    "# load program\n",
    "module load blast\n",
    "\n",
    "# assign variables\n",
    "GenomeDir=\"/scratch/user/uqpobri2/data/sponge_symbionts\"\n",
    "DataDir=\"/scratch/user/uqpobri2/data/sponge_symbionts/taurine\"\n",
    "OutDir=\"/scratch/user/uqpobri2/analysis/sponge_symbionts/blast\"\n",
    "\n",
    "mkdir -p $OutDir/db\n",
    "mkdir -p $OutDir/taurine\n",
    "\n",
    "# check variables\n",
    "echo \"\"\n",
    "echo \"#====================#\"\n",
    "echo \"#===== JOB INFO =====#\"\n",
    "echo \"#====================#\"\n",
    "echo \"\"\n",
    "echo \"    SLURM Job ID      :  ${SLURM_JOB_ID}\"\n",
    "echo \"    SLURM Job Name    :  ${SLURM_JOB_NAME}\"\n",
    "echo \"\"\n",
    "echo \"    Input genome path :  ${GenomeDir}\"\n",
    "echo \"    Input sequences   :  ${DataDir}\" \n",
    "echo \"    Output directory  :  ${OutDir}\"\n",
    "echo \"\"\n",
    "echo \"#====================#\"\n",
    "echo \"\"\n",
    "\n",
    "## blast to genomes\n",
    "\n",
    "# concatenate genomes from each family to one file\n",
    "for Dir in $(ls $GenomeDir | grep -v taurine); do\n",
    "    # get dir name\n",
    "    echo $Dir\n",
    "   # concatenate files in dir\n",
    "    cat $GenomeDir/${Dir}/*.fna > $GenomeDir/${Dir}_all.fna\n",
    "done\n",
    "\n",
    "# make symbiont genome database\n",
    "for File in $(ls $GenomeDir/*all.fna); do\n",
    "    # check file name\n",
    "    echo $File\n",
    "    # get symbiont name\n",
    "    Name=$(basename $File | cut -f 1 -d '_')\n",
    "    # check name\n",
    "    echo $Name\n",
    "    #make blast db\n",
    "   srun makeblastdb \\\n",
    "        -in $File \\\n",
    "        -dbtype nucl \\\n",
    "        -out $OutDir/db/$Name/$Name\n",
    "done\n",
    "\n",
    "# tblastx to genome database (translated protein search)\n",
    "for db in $(ls $OutDir/db); do\n",
    "    #check db\n",
    "    echo $db\n",
    "    # blast sequences to db \n",
    "    srun tblastx \\ \n",
    "        -query $DataDir/tauABC.fasta \\\n",
    "        -db $OutDir/db/$db/$db \\\n",
    "        -out $OutDir/taurine/taurine_tblastx_results_${db}.txt \\\n",
    "        -outfmt 6\n",
    "done\n"
   ]
  },
  {
   "cell_type": "raw",
   "id": "34f3a1b2-70c8-46f1-ac46-aebe71d540e3",
   "metadata": {},
   "source": [
    "The columns in the tabular output file will include:\n",
    "\n",
    "1 Query sequence ID\n",
    "2 Subject sequence ID\n",
    "3 Percent identity\n",
    "4 Alignment length\n",
    "5 Number of mismatches\n",
    "6 Number of gap openings\n",
    "7 Start position in query sequence\n",
    "8 End position in query sequence\n",
    "9 Start position in subject sequence\n",
    "10 End position in subject sequence\n",
    "11 E-value\n",
    "12 Bit score"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "28b27957-af52-4d25-8229-5b18466f6e76",
   "metadata": {},
   "outputs": [],
   "source": [
    "# to view: sort by bit score - 12,12nr - reflects overall quality - higher the better\n",
    "# note: nr = numeric, reverse order\n",
    "sort -k 12,12nr $OutDir/taurine/taurine_blast_results_micro.txt | tsv_view | less -S\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fe0586d3-758f-4dd7-9022-945f6dd874e1",
   "metadata": {},
   "source": [
    "### Find which genomes have contigs that have a signficant match to tauABC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "55631b93-0976-4627-ba71-017ee5782b5f",
   "metadata": {},
   "outputs": [],
   "source": [
    "# output significant results, in this case percent identity > 30, aligment >100, e-value < 1e-03, bit score >50\n",
    "# based on Pearson 2013\n",
    "\n",
    "# from output dir..\n",
    "for File in *tblastx*; do\n",
    "    # get sample id\n",
    "    echo $File\n",
    "    SampleID=$(echo $File | cut -d '_' -f 4 )\n",
    "    echo $SampleID\n",
    "   # get significant results\n",
    "    awk '$3 > 30 && $4 > 100 && $11 < 1e-03 && $12 > 50 {print}' $File | sort -k 12,12nr > taurine_tblastx_significant_results_e03-100_${SampleID}\n",
    "done\n",
    "\n",
    "# get unique gene/contig matches\n",
    "for File in *tblastx_significant_results*; do\n",
    "    # get sample id\n",
    "    echo $File\n",
    "    SampleID=$(echo $File | cut -d '_' -f 6 )\n",
    "    echo $SampleID\n",
    "    # output unique matches\n",
    "    awk '{print $1, $2}' $File | sort | uniq > taurine_tblastx_unique_matches_e03-100_${SampleID}\n",
    "done\n",
    "\n",
    "# get genomes for matched contigs\n",
    "# running each separately to avoid looping issues\n",
    "# endo\n",
    "while read -r line; do\n",
    "    col1=$(echo \"$line\" | awk '{print $1}')\n",
    "    col2=$(echo \"$line\" | awk '{print $2}')\n",
    "    match_found=false\n",
    "\n",
    "    for file in /scratch/user/uqpobri2/data/sponge_symbionts/endo/*; do\n",
    "        GenomeID=$(basename \"$file\")\n",
    "        if grep -q \"$col2\" \"$file\"; then\n",
    "            echo \"$col1 $col2 $GenomeID\" >> genomes_with_taurine_e03-100_endo.txt\n",
    "            match_found=true\n",
    "            break\n",
    "        fi\n",
    "    done\n",
    "\n",
    "    if [ \"$match_found\" = false ]; then\n",
    "        echo \"$col1 $col2\" >> genomes_with_taurine_e03-100_endo.txt\n",
    "    fi\n",
    "done < taurine_tblastx_unique_matches_e03-100_endo.txt \n",
    "\n",
    "# micro\n",
    "while read -r line; do\n",
    "    col1=$(echo \"$line\" | awk '{print $1}')\n",
    "    col2=$(echo \"$line\" | awk '{print $2}')\n",
    "    match_found=false\n",
    "\n",
    "    for file in /scratch/user/uqpobri2/data/sponge_symbionts/micro/*; do\n",
    "        GenomeID=$(basename \"$file\")\n",
    "        if grep -q \"$col2\" \"$file\"; then\n",
    "            echo \"$col1 $col2 $GenomeID\" >> genomes_with_taurine_e03-100_micro.txt\n",
    "            match_found=true\n",
    "            break\n",
    "        fi\n",
    "    done\n",
    "\n",
    "    if [ \"$match_found\" = false ]; then\n",
    "        echo \"$col1 $col2\" >> genomes_with_taurine_e03-100_micro.txt\n",
    "    fi\n",
    "done < taurine_tblastx_unique_matches_e03-100_micro.txt\n",
    "\n",
    "# nitro\n",
    "while read -r line; do\n",
    "    col1=$(echo \"$line\" | awk '{print $1}')\n",
    "    col2=$(echo \"$line\" | awk '{print $2}')\n",
    "    match_found=false\n",
    "\n",
    "    for file in /scratch/user/uqpobri2/data/sponge_symbionts/nitro/*; do\n",
    "        GenomeID=$(basename \"$file\")\n",
    "        if grep -q \"$col2\" \"$file\"; then\n",
    "            echo \"$col1 $col2 $GenomeID\" >> genomes_with_taurine_e03-100_nitro.txt\n",
    "            match_found=true\n",
    "            break\n",
    "        fi\n",
    "    done\n",
    "\n",
    "    if [ \"$match_found\" = false ]; then\n",
    "        echo \"$col1 $col2\" >> genomes_with_taurine_e03-100_nitro.txt\n",
    "    fi\n",
    "done < taurine_tblastx_unique_matches_e03-100_nitro.txt\n",
    "\n",
    "# spiro\n",
    "while read -r line; do\n",
    "    col1=$(echo \"$line\" | awk '{print $1}')\n",
    "    col2=$(echo \"$line\" | awk '{print $2}')\n",
    "    match_found=false\n",
    "\n",
    "    for file in /scratch/user/uqpobri2/data/sponge_symbionts/spiro/*; do\n",
    "        GenomeID=$(basename \"$file\")\n",
    "        if grep -q \"$col2\" \"$file\"; then\n",
    "            echo \"$col1 $col2 $GenomeID\" >> genomes_with_taurine_e03-100_spiro.txt\n",
    "            match_found=true\n",
    "            break\n",
    "        fi\n",
    "    done\n",
    "\n",
    "    if [ \"$match_found\" = false ]; then\n",
    "        echo \"$col1 $col2\" >> genomes_with_taurine_e03-100_spiro.txt\n",
    "    fi\n",
    "done < taurine_tblastx_unique_matches_e03-100_spiro.txt\n",
    "\n",
    "# thermo\n",
    "while read -r line; do\n",
    "    col1=$(echo \"$line\" | awk '{print $1}')\n",
    "    col2=$(echo \"$line\" | awk '{print $2}')\n",
    "    match_found=false\n",
    "\n",
    "    for file in /scratch/user/uqpobri2/data/sponge_symbionts/thermo/*; do\n",
    "        GenomeID=$(basename \"$file\")\n",
    "        if grep -q \"$col2\" \"$file\"; then\n",
    "            echo \"$col1 $col2 $GenomeID\" >> genomes_with_taurine_e03-100_thermo.txt\n",
    "            match_found=true\n",
    "            break\n",
    "        fi\n",
    "    done\n",
    "\n",
    "    if [ \"$match_found\" = false ]; then\n",
    "        echo \"$col1 $col2\" >> genomes_with_taurine_e03-100_thermo.txt\n",
    "    fi\n",
    "done < taurine_tblastx_unique_matches_e03-100_thermo.txt\n",
    "\n",
    "# get unique gene/genome matches\n",
    "for File in *with_taurine*; do\n",
    "    # get sample id\n",
    "    echo $File\n",
    "    SampleID=$(echo $File | cut -d '_' -f 5 )\n",
    "    echo $SampleID\n",
    "    # output unique matches\n",
    "    awk '{print $1, $3}' $File | sort | uniq > genomes_with_taurine_unique_matches_e03-100_${SampleID}\n",
    "done\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "188ea5a2-1607-47a1-ac22-1e3d0d3d35c3",
   "metadata": {},
   "outputs": [],
   "source": [
    "# note: genes lengths for tauABC\n",
    "# tauA = 980\n",
    "# tauB = 781\n",
    "# tauC = 842"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
